pkg/operator/ceph/object/user/controller.go

```diff
@@ -305,19 +307,24 @@ func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectSt
 	}
 
 	// Update caps if necessary
+	user.UserCaps = generateUserCaps(user)
 	if user.UserCaps != r.userConfig.UserCaps {
 		// If they are no caps to be removed, the API will return an error "missing user capabilities"
 		if user.UserCaps != "" {
+			logger.Tracef("remove capabilities %s from user %s", user.UserCaps, r.userConfig.ID)
 			_, err = r.objContext.AdminOpsClient.RemoveUserCap(r.opManagerContext, r.userConfig.ID, user.UserCaps)
 			if err != nil {
 				return errors.Wrapf(err, "failed to remove current ceph object user %q capabilities", r.userConfig.ID)
 			}
 		}
-		_, err = r.objContext.AdminOpsClient.AddUserCap(r.opManagerContext, r.userConfig.ID, r.userConfig.UserCaps)
-		if err != nil {
-			return errors.Wrapf(err, "failed to update ceph object user %q capabilities", r.userConfig.ID)
-		}
 		logCreateOrUpdate = fmt.Sprintf("updated ceph object user %q", u.Name)
```

Comment:
Sorry what I meant was the end of the condition `user.UserCaps != r.userConfig.UserCaps`

Reply:
I keep the location of this log similar to the previous one, that is, add the log after setting user caps